src/hiveboard_bridge/test/SocketToMoveByIntegrationTest.cpp

```diff
@@ -0,0 +1,78 @@
+#include "ros/ros.h"
+#include <arpa/inet.h>
+#include <chrono>
+#include <common/IProtobufStream.h>
+#include <cstdint>
+#include <cstring>
+#include <hivemind-host/HiveMindHostSerializer.h>
+#include <memory>
+#include <netinet/in.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <thread>
+#include <unistd.h>
+
+class TCPClient : public IProtobufStream {
+  public:
+    TCPClient() { m_clientFd = ::socket(AF_INET, SOCK_STREAM, 0); }
+
+    int connect() {
+        struct sockaddr_in serv_addr;
+
+        serv_addr.sin_family = AF_INET;
+        serv_addr.sin_port = htons(8080);
+
+        // Convert IPv4 and IPv6 addresses from text to binary form
+        if (inet_pton(AF_INET, "127.0.0.1", &serv_addr.sin_addr) <= 0) {
+            printf("\nInvalid address/ Address not supported \n");
+            return -1;
+        }
+
+        ::connect(m_clientFd, (struct sockaddr*)&serv_addr, sizeof(serv_addr));
+    }
+
+    bool receive(uint8_t* data, uint16_t length) override {
+        int nbBytesReceived = ::recv(m_clientFd, data, length, MSG_WAITALL);
+
+        return nbBytesReceived == length;
+    }
+
+    bool send(const uint8_t* data, uint16_t length) override {
+        if (::send(m_clientFd, data, length, 0) > -1)
+            return true;
+        else
+            return false;
+    }
+
+  private:
+    int m_clientFd;
+};
```

Comment:
I'm not sure why is this class in the test and not defined in it's own file, why is it not the same code for the main and this test?

Reply:
The "integration test" is not really self-contained. It is really a secondary node that needs the `hiveboard_bridge` to be executed.
I would go so far as to say that the integration test node is simply a sandbox that mocks the HiveBoard (it has a TCP a client, whilst the bridge is a server).

Reply:
I could move the TCP client class in another file, but just keep in mind that this is all just a little sandbox to test stuff.